src/dataframerow/dataframerow.jl

```diff
@@ -17,7 +17,7 @@ function Base.getindex(r::DataFrameRow, idx::AbstractArray)
     return DataFrameRow(parent(r)[idx], row(r))
 end
 
-function Base.getindex(r::DataFrameRow, idx::Any)
+function Base.getindex(r::DataFrameRow, idx::ColumnIndex)
```

Comment:
Does this need a deprecation?

Reply:
The only case offending was when `idx` was a colon, where a `DataFrame` was returned. I have fixed it (but we return `DataFrameRow` now as I guess this is more consistent, and the old behavior was unintended). We can add deprecation for it if you prefer.